### PR TITLE
auth providers will return entity ref as "sub" value

### PR DESCRIPTION
**Pull request by Edje-C**
## Hey, I just made a Pull Request!
Returning entity ref instead of id as auth provider sub value.

#### :heavy_check_mark: Checklist

- [ ] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [ ] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Review by Rugvip — COMMENTED** [review]
Thank you! :grin:

Please make sure the checks are green, there's more information about DCO in the details link, and it looks like one of the tests need updating.

This change will also need a [changeset](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets). I think it's best me make it a `minor` (breaking) change since we're changing the behavior of the API after all, even if it's handled by the recent frontend additions.

File: plugins/auth-backend/src/providers/okta/provider.ts [hunk 1]
```diff
@@ -219,7 +219,7 @@ export const oktaEmailSignInResolver: SignInResolver<OAuthResult> = async (
   const claims = getEntityClaims(entity);
   const token = await ctx.tokenIssuer.issueToken({ claims });
 
-  return { id: entity.metadata.name, entity, token };
+  return { id: `user:default/${entity.metadata.name}`, entity, token };
```
Comment (freben) [inline plugins/auth-backend/src/providers/okta/provider.ts:222]: remove

File: plugins/auth-backend/src/providers/bitbucket/provider.ts [hunk 2]
```diff
@@ -222,7 +222,7 @@ export const bitbucketUsernameSignInResolver: SignInResolver<BitbucketOAuthResul
     const claims = getEntityClaims(entity);
     const token = await ctx.tokenIssuer.issueToken({ claims });
 
-    return { id: entity.metadata.name, entity, token };
+    return { id: `user:default/${entity.metadata.name}`, entity, token };
```
Comment (freben) [inline plugins/auth-backend/src/providers/bitbucket/provider.ts:225]: Actually, you can remove this change. The `id` is deprecated and should not be used at all, but it used to have the documented semantics that it should be "for example the metadata.name". There may be old code out there that relies on it. The important thing is that the `sub` and `ent` ones are full refs! Those fixes are good.

File: plugins/auth-backend/src/providers/bitbucket/provider.ts [hunk 3]
```diff
@@ -242,7 +242,7 @@ export const bitbucketUserIdSignInResolver: SignInResolver<BitbucketOAuthResult>
...
```
Comment (freben) [inline plugins/auth-backend/src/providers/bitbucket/provider.ts:245]: remove

**Review by freben — COMMENTED** [review]
Hey I'm sorry I didn't spot this upfront - we actually only need the `sub` to be full refs, and the old deprecated `id` can be left as is. Again sorry for the confusion, and thanks for helping out :)

**Pull request closed without merging**